src/coreclr/jit/emitarm64.cpp

```diff
@@ -8976,6 +9067,99 @@ void emitter::emitIns_R_R_I(instruction     ins,
     appendToCurIG(id);
 }
 
+/*****************************************************************************
+ *
+ *  Add an instruction referencing two registers and a floating point constant.
+ */
+
+void emitter::emitIns_R_R_F(
+    instruction ins, emitAttr attr, regNumber reg1, regNumber reg2, double immDbl, insOpts opt /* = INS_OPTS_NONE */)
+{
+    ssize_t   imm  = 0;
+    emitAttr  size = EA_SIZE(attr);
+    insFormat fmt  = IF_NONE;
+
+    bool      canEncode = false;
+    floatImm8 fpi;
+
+    /* Figure out the encoding format of the instruction */
+    switch (ins)
+    {
+        case INS_sve_fmul:
+        case INS_sve_fmaxnm:
+        case INS_sve_fadd:
+        case INS_sve_fmax:
+        case INS_sve_fminnm:
+        case INS_sve_fsub:
+        case INS_sve_fmin:
+        case INS_sve_fsubr:
+#ifdef DEBUG
+            switch (ins)
+            {
+                case INS_sve_fadd:
+                case INS_sve_fsub:
+                case INS_sve_fsubr:
+                    assert((immDbl == 0.5) || (immDbl == 1.0));
+                    break;
+
+                case INS_sve_fmax:
+                case INS_sve_fmaxnm:
+                case INS_sve_fmin:
+                case INS_sve_fminnm:
+                    assert((immDbl == 0) || (immDbl == 1.0));
+                    break;
+
+                case INS_sve_fmul:
+                    assert((immDbl == 0.5) || (immDbl == 2.0));
+                    break;
+
+                default:
+                    assert(!"Invalid instruction");
+                    break;
+            }
+#endif // DEBUG
+
+            assert(insOptsScalableAtLeastHalf(opt));
+            assert(isVectorRegister(reg1));
+            assert(isLowPredicateRegister(reg2));
+            assert(isScalableVectorSize(size));
+
+            if (immDbl != 0.0)
+            {
+                fpi.immFPIVal = 0;
```

Comment:
I'm not super happy about it, but this is how we handle storing immediate floats in `instrDesc`. The current encode and decode functions are complicated, but using them isn't so bad.

For the `90` and `270` values, they are just like any other immediate value so I don't think we need to change that.

Reply:
Also I think here, we should encode them as `0`, `1`, etc. Related: https://github.com/dotnet/runtime/pull/98187#discussion_r1484091812

Reply:
I am wondering why we need special handling for `immDbl == 0.0`? doesn't `canEncodeFloatImm8()` handle it?

Reply:
It doesn't handle `0` which is why I had to do it myself. However, since we are going to encode the rotation values as `0` - `3`, I guess we should do the same here.